### PR TITLE
chore: upgrade rust toolchain to 1.72

### PR DESCRIPTION
**Pull request by KaoImin**
## What this PR does / why we need it?

This PR upgrades rust toolchain to 1.72. The mainly reason is:
> [Rust reports potentially useful `cfg-disabled` items in errors](https://blog.rust-lang.org/2023/08/24/Rust-1.72.0.html#rust-reports-potentially-useful-cfg-disabled-items-in-errors)

### What is the impact of this PR?

No Breaking Change

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [x] E2E Tests
- [x] Web3 Compatible Tests
- [x] OCT 1-5 And 12-15
- [x] OCT 6-10
- [x] OCT 11
- [x] OCT 16-19
- [x] v3 Core Tests

### **CI Description**

| CI Name                                   | Description                                                               |
| ----------------------------------------- | ------------------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`                     |
| *Coverage Test*                           | Get the unit test coverage report                                         |
| *E2E Test*                                | Run end-to-end test to check interfaces                                   |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`           |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                                       |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3                        |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin                      |

</details>

### CODE REVIEW

**Review by Flouse — CHANGES_REQUESTED** [review]
Maybe update Dockerfile too?
https://github.com/axonweb3/axon/blob/c86f0658474bc60012a051c8a4c123959691222d/Dockerfile#L1

File: core/mempool/src/adapter/mod.rs [hunk 1]
```diff
@@ -639,7 +639,7 @@ mod tests {
             M: MessageCodec,
             P: AsRef<[Bytes]> + Send + 'a,
         {
-            unreachable!()
+            Ok(())
```
Comment (blckngm) [inline core/mempool/src/adapter/mod.rs:642]: Looks like this is the only non-trivial change.

**Review by blckngm — APPROVED** [review]

File: core/mempool/src/adapter/mod.rs (hunk 1 above)
Comment (KaoImin) [reply core/mempool/src/adapter/mod.rs:642]: This is required by `clippy`.

Comment (Flouse) [reply core/mempool/src/adapter/mod.rs:642]: Shall we add a comment to make `clippy` silence here, avoid changing any Axon logic in this PR?

Comment (KaoImin) [reply core/mempool/src/adapter/mod.rs:642]: The code is in unit test.

**Review by Flouse — APPROVED** [review]

File: core/executor/src/tests/system_script/image_cell.rs [hunk 2]
```diff
@@ -60,10 +60,7 @@ fn test_update_first<'a>(
 
     let root = backend.storage(CKB_LIGHT_CLIENT_CONTRACT_ADDRESS, *HEADER_CELL_ROOT_KEY);
     let cell_key = CellKey::new([7u8; 32], 0x0);
-    let get_cell = ImageCellReader::default()
-        .get_cell(root, &cell_key)
-        .unwrap()
```
Comment (distributedstatemachine) [inline core/executor/src/tests/system_script/image_cell.rs:65]: Do you think it makes sense to handle this error gracefully via ? or map_err?

File: core/network/src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#![allow(clippy::uninlined_format_args, dead_code)]
+#![allow(dead_code)]
```
Comment (distributedstatemachine) [inline core/network/src/lib.rs:1]: Why do we want to allow dead code here ?

File: protocol/src/codec/transaction.rs [hunk 4]
```diff
@@ -362,59 +362,14 @@ impl Decodable for SignedTransaction {
...
     use common_crypto::secp256k1_recover;
 
     use crate::codec::hex_decode;
     use crate::types::{
         AddressSource, Bytes, CKBTxMockByRefAndOneInput, CellDep, CellWithData, Public, Script,
-        SignatureR, SignatureS, TransactionAction, Witness, H160, H256, U256,
+        SignatureR, SignatureS, Witness, H160, H256, U256,
     };
 
-    fn rand_bytes(len: usize) -> Bytes {
```
Comment (distributedstatemachine) [inline protocol/src/codec/transaction.rs:375]: Why is this test removed ? is there a rplace for it ? or is it no longer relevant?

File: core/executor/src/tests/system_script/image_cell.rs (hunk 2 above)
Comment (KaoImin) [reply core/executor/src/tests/system_script/image_cell.rs:65]: The code is in unit test. If error occur, panic at once.

File: protocol/src/codec/transaction.rs (hunk 4 above)
Comment (KaoImin) [reply protocol/src/codec/transaction.rs:375]: They are some assist functions that do not used by unit tests.

File: core/network/src/lib.rs (hunk 3 above)
Comment (KaoImin) [reply core/network/src/lib.rs:1]: Most of this dead code is some configuration item. Initially they could be configured, and some of them were removed during development. We thought that these configurations might be reintroduced later, so we didn't remove them.

**Review by distributedstatemachine — APPROVED** [review]

**Pull request merged**